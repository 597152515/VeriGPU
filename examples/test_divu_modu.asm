li x1, 4
li x2, 4
divu x3, x1, x2
remu x4, x1, x2
outr x3  ; 1 r 0
outr x4

li x1, 4
li x2, 2
divu x3, x1, x2
remu x4, x1, x2
outr x3  ; 2 r 0
outr x4

li x1, 0
li x2, 0
divu x3, x1, x2
remu x4, x1, x2
outr x3  ; ffffffff r 0
outr x4

li x1, 0
li x2, 2
divu x3, x1, x2
remu x4, x1, x2
outr x3  ; 0 r 0
outr x4

li x1, 2222
li x2, 123
divu x3, x1, x2
remu x4, x1, x2
outr x3  ; 18 r 8
outr x4

li x1, 1111
li x2, 444
divu x3, x1, x2
remu x4, x1, x2
outr x3  ; 2 r 223
outr x4

halt
