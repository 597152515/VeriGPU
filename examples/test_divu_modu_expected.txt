out 0 00000000000000000000000000010010 00000012 18
out 1 00000000000000000000000000001000 00000008 8
out 2 00000000000000000000000000000010 00000002 2
out 3 00000000000000000000000011011111 000000df 223
